services/src/main/java/org/keycloak/protocol/oid4vc/model/VerifiableCredential.java

```diff
@@ -87,6 +98,29 @@ public VerifiableCredential setIssuer(URI issuer) {
         return this;
     }
 
+    public VerifiableCredential setIssuer(URL issuer) {
```

Comment:
Even thought the spec mandates the usage of URL, the usage of the java URL class would not allow did's as issuers anymore. While a did does not always fulfil the requriements of a URL, a did:web for example would. I would rather not use URL here.

Reply:
I just added an additional method. The overloaded URI method is still present. If we look at the examples from w3c URLs seem desirable in case of issuers:

```json
{
  "@context": [
    "https://www.w3.org/ns/credentials/v2",
    "https://www.w3.org/ns/credentials/examples/v2"
  ],
  "id": "http://university.example/credentials/3732",
  "type": ["VerifiableCredential", "ExampleDegreeCredential"],
  "issuer": "https://university.example/issuers/565049",
  "validFrom": "2010-01-01T00:00:00Z",
  "credentialSubject": {
    "id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
    "degree": {
      "type": "ExampleBachelorDegree",
      "name": "Bachelor of Science and Arts"
    }
  }
}
```

I can remove the method if necessary I just thought it doesn't hurt to have both added `URI` and `URL`

Reply:
Needed to change this completely because the serialization of the object has become problematic